apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java

```diff
@@ -424,6 +433,58 @@ public String fetchTimersInfo(){
         return Action.ERROR.toUpperCase();
     }
 
+    private List<Integer> reduceApiCollectionToId(List<ApiCollection> apiCollections) {
+        if (apiCollections == null) {
+            return new ArrayList<>();
+        }
+        return apiCollections.stream().map(apiCollection -> apiCollection.getId()).collect(Collectors.toList());
+    }
+
+    private List<ApiCollection> filterDeactivatedCollections(List<ApiCollection> apiCollections) {
+        if (apiCollections == null) {
+            return new ArrayList<>();
+        }
+        List<Integer> apiCollectionIds = reduceApiCollectionToId(this.apiCollections);
+        /*
+         * The apiCollections from request contain only the IDs,
+         * thus we need to fetch the active status from the db.
+         */
+        return ApiCollectionsDao.instance.findAll(Filters.and(
+                Filters.in(Constants.ID, apiCollectionIds),
+                Filters.eq(ApiCollection._DEACTIVATED, true)));
+    }
+
+    public String deactivateCollections() {
```

Comment:
Show a confirmation dialog - mention that we will stop ingesting API data for these collections. And no tests will be allowed.